nomad/state/state_store.go

```diff
@@ -1080,6 +1079,70 @@ func deleteNodeFromPlugin(txn *memdb.Txn, plug *structs.CSIPlugin, node *structs
 	return nil
 }
 
+// deleteJobFromPlugin removes allocations from this job from the plugins those jobs are
+// running, possibly deleting the plugin. It's called in DeleteJobTxn
+func (s *StateStore) deleteJobFromPlugin(index uint64, txn *memdb.Txn, job *structs.Job) error {
+	ws := memdb.NewWatchSet()
+	allocs, err := s.AllocsByJob(ws, job.Namespace, job.ID, false)
+	if err != nil {
+		return fmt.Errorf("error getting allocations: %v", err)
+	}
+
+	type pair struct {
+		p string
```

Comment:
I know we're going for brevity here but let's call this a `pluginID` (or at least `pID` or something) so that it's clear what we've got -- there's "pairs" "plugAllocs" and "plugs" all running around at the same time in this function and they all start with "p"! 😀

Reply:
agreed!